### PR TITLE
Initial

### PR DESCRIPTION
**Pull request by talyguryn**
```
$ yarn build

🗄 Recreating output directory
🔎 Getting SVG files
🛠 Processing SVG files
💎 arrow-down.svg -> IconArrowDown
💎 arrow-left.svg -> IconArrowLeft
💎 arrow-right.svg -> IconArrowRight
💎 arrow-up.svg -> IconArrowUp
💎 bold.svg -> IconBold
💎 checklist.svg -> IconChecklist
💎 file.svg -> IconFile
💎 gift.svg -> IconGift
💎 h1.svg -> IconH1
💎 h2.svg -> IconH2
💎 h3.svg -> IconH3
💎 h4.svg -> IconH4
💎 h5.svg -> IconH5
💎 h6.svg -> IconH6
💎 inline-code.svg -> IconInlineCode
💎 italic.svg -> IconItalic
💎 link.svg -> IconLink
💎 list-bulleted.svg -> IconListBulleted
💎 list-numbered.svg -> IconListNumbered
💎 marker.svg -> IconMarker
💎 menu.svg -> IconMenu
💎 play.svg -> IconPlay
💎 plus.svg -> IconPlus
💎 search.svg -> IconSearch
💎 star.svg -> IconStar
💎 table.svg -> IconTable
💎 text.svg -> IconText
📝 Updating the README.md file
📊 27 of 27 icons were generated
```

### CODE REVIEW

File: bin/generate.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+const files = fs.readdirSync('./src/icons');
+
+fs.writeFileSync(ICONS_BUNDLE_FILE, '');
+
+const TABLE_OF_ICONS = [
+    '| Icon | Name |',
+    '| --- | --- |'
+];
+
+for (let filename of files) {
```
Comment (neSpecc) [inline bin/generate.js:19]: move the processing of the single file to a separate method.

File: bin/generate.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        plugins: [
+            'sortAttrs',
+            {
+                name: 'addAttributesToSVGElement',
+                params: {
+                    attributes: [
+                        {'fill': 'currentColor'}
+                    ]
+                }
+            }
```
Comment (neSpecc) [inline bin/generate.js:35]: also, we need to pass `prefer viewBox` (can't remember the exact name of the option)

File: bin/generate.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        ]
+    });
+    const optimizedSvgString = result.data;
+
+    fs.writeFileSync(`./dist/icons/${exportName}.svg`, optimizedSvgString);
```
Comment (neSpecc) [inline bin/generate.js:40]: you're repeating the dist path several times. Lets move it to the variable

File: bin/generate.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+const TABLE_OF_ICONS = [
+    '| Icon | Name |',
+    '| --- | --- |'
+];
+
+for (let filename of files) {
+    const name = filename.split('.')[0];
+    const exportName = `Icon${name}`;
+
+    let iconCode = fs.readFileSync(`./src/icons/${filename}`, 'utf8');
```
Comment (neSpecc) [inline bin/generate.js:23]: I think we need to use the `path` package for the all path strings to make sure that it will work correctly on every OS

**Review by neSpecc — CHANGES_REQUESTED** [review]

File: .github/workflows/build-icons.yml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+jobs:
+  lint:
+    name: Build frontend
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js 15
+        uses: actions/setup-node@v1
+        with:
+          node-version: 15
```
Comment (ilyamore88) [inline .github/workflows/build-icons.yml:14]: Why not LTS (16) or latest (18)?

**Comment by talyguryn** [comment]
image

**Review by ilyamore88 — APPROVED** [review]

**Review by neSpecc — APPROVED** [review]

**Pull request merged**